### DISCUSSION TITLE
Load policies from string

### DISCUSSION THREAD

**Discussion by nosilex** [microcosm-cc/bluemonday#177] [Ideas]
Hello everyone, what do you think about the possibility of loading policies from a string? That way it would be more flexible to make dynamic adjustments without having to build the application.

An example would be like this:
```json
{
  "AllowElements": ["b","i","li","ul","ol","mark","em","strong"]
  "AllowAttrs":[
    {
      "element": "a",
      "attributes": ["href","target"]
    },
    {
      "element": "p",
      "attributes": ["class"]
    }
  ],
  "RequireParseableURLs":true,
  "AllowURLSchemes":["mailto", "http", "https"],
  "AddTargetBlankToFullyQualifiedLinks": true
}
```

Any suggestion is welcome.

**Comment by grafana-dee** [comment]
I like this... this is a good idea.

I've seen that bluemonday is included in a lot of things like Hugo, so giving the ability to configure it dynamically and not require recompilation is good.

I'll have to think about the format of that string... and TBF any software could've implemented this already, but then if I make this more standard it will encourage making that config available when people integrate bluemonday.

**Reply by nosilex** [reply]
Excellent!

Can you help me in defining the string format? Then I can participate in the development of the feature.
